pymtl3/passes/sverilog/import_/ImportPass.py

```diff
@@ -105,100 +119,108 @@ def get_imported_object( s, m ):
+  def create_verilator_model( s, m, config, cached ):
+    """Verilate module `m`."""
+    config.check_options()
+    config.vprint("\n=====Verilate model=====")
     if not cached:
-      obj_dir = 'obj_dir_' + top_name
-      flags = [
-        '--unroll-count 1000000', '--unroll-stmts 1000000', '--assert',
-        '-Wno-UNOPTFLAT', '-Wno-UNSIGNED', ]
-      if dump_vcd:
-        flags.append( "--trace" )
-      flags = " ".join( flags )
-      cmd = \
-"""\
-verilator -cc {sv_file_path} -top-module {top_name} --Mdir {obj_dir} -O3 {flags}
-"""
-      cmd = cmd.format( **locals() )
+      # Generate verilator command
+      cmd = config.create_vl_cmd()
 
       # Remove obj_dir directory if it already exists.
       # obj_dir is where the verilator output ( C headers and sources ) is stored
+      obj_dir = config.get_vl_mk_dir()
       if os.path.exists( obj_dir ):
         shutil.rmtree( obj_dir )
 
-      # Print out the modification time stamp of SystemVerilog source file
-      # print 'Modification timestamp of {}.sv: {}'.format(
-          # top_name, os.path.getmtime( top_name + '.sv' ) )
-
       # Try to call verilator
       try:
+        config.vprint(f"Verilating {config.get_top_module()} with command:", 2)
+        config.vprint(f"{cmd}", 4)
         subprocess.check_output( cmd, stderr = subprocess.STDOUT, shell = True )
       except subprocess.CalledProcessError as e:
-        assert False, \
-"""\
-Fail to verilate model {} in file {}
-  Verilator command:
-  {}
+        err_msg = e.output if not isinstance(e.output, bytes) else \
+                  e.output.decode('utf-8')
+        import_err_msg = \
+            f"Fail to verilate model {config.get_option('top_module')}\n"\
+            f"  Verilator command:\n{indent(cmd, '  ')}\n\n"\
+            f"  Verilator output:\n{indent(wrap(err_msg), '  ')}\n"
+        raise SVerilogImportError(m, import_err_msg) from e
+      config.vprint(f"Successfully verilated the given model!", 2)
 
-  Verilator output:
-  {}
-""".format( top_name, sv_file_path, cmd, e.output )
+    else:
+      config.vprint(f"{config.get_top_module()} not verilated because it's cached!", 2)
 
   #-----------------------------------------------------------------------
   # create_verilator_c_wrapper
   #-----------------------------------------------------------------------
 
-  def create_verilator_c_wrapper( s, m, full_name, packed_ports, dump_vcd, cached ):
+  def create_verilator_c_wrapper( s, m, config, packed_ports, cached ):
     """Return the file name of generated C component wrapper.
 
     Create a C wrapper that calls verilator C API and provides interfaces
     that can be later called through CFFI.
     """
-    component_name = full_name
-    try:
-      vcd_timescale = m.vcd_timescale
-    except AttributeError:
-      vcd_timescale = "10ps"
+    p_map = config.get_port_map() if config.is_port_mapped() else lambda x: x
+    component_name = config.get_top_module()
```

Comment:
Did you implement the 2-level caching we discussed in this PR?

Reply:
No not in this PR.